symphony/bdk/core/service/message/message_service.py

```diff
@@ -0,0 +1,340 @@
+from symphony.bdk.gen.agent_model.v4_message_blast_response import V4MessageBlastResponse
+from symphony.bdk.gen.agent_model.v4_message_import_list import V4MessageImportList
+from symphony.bdk.gen.model_utils import (
+    file_type
+)
+from symphony.bdk.gen.pod_api.default_api import DefaultApi
+from symphony.bdk.gen.pod_api.message_api import MessageApi
+from symphony.bdk.gen.pod_api.message_suppression_api import MessageSuppressionApi
+from symphony.bdk.gen.pod_api.pod_api import PodApi
+from symphony.bdk.gen.pod_api.streams_api import StreamsApi
+from symphony.bdk.gen.pod_model.message_metadata_response import MessageMetadataResponse
+from symphony.bdk.gen.pod_model.message_receipt_detail_response import MessageReceiptDetailResponse
+from symphony.bdk.gen.pod_model.message_status import MessageStatus
+from symphony.bdk.gen.pod_model.message_suppression_response import MessageSuppressionResponse
+from symphony.bdk.gen.pod_model.stream_attachment_item import StreamAttachmentItem
+
+
+class MessageService:
+    """Service class for managing messages.
+    """
+
+    def __init__(self, messages_api: MultiAttachmentsMessagesApi,
+                 message_api: MessageApi,
+                 message_suppression_api: MessageSuppressionApi,
+                 streams_api: StreamsApi,
+                 pod_api: PodApi,
+                 attachment_api: AttachmentsApi,
+                 default_api: DefaultApi,
+                 auth_session: AuthSession
+                 ):
+        self._messages_api = messages_api
+        self._message_api = message_api
+        self._message_suppression_api = message_suppression_api
+        self._streams_api = streams_api
+        self._pod_api = pod_api
+        self._attachment_api = attachment_api
+        self._default_api = default_api
+        self._auth_session = auth_session
+
+    async def list_messages(
+            self,
+            stream_id: str,
+            since: int = 0,
+            skip: int = 0,
+            limit: int = 50
+    ) -> [V4Message]:
+        """Get messages from an existing stream. Additionally returns any attachments associated with the message.
+        See: `Messages <https://developers.symphony.com/restapi/reference#messages-v4>`_
+
+        :param stream_id: The stream where to look for messages
+        :param since: Timestamp of the earliest possible date of the first message returned.
+        :param skip: Number of messages to skip. Default: 0
+        :param limit: Maximum number of messages to return. Default: 0
+
+        :return: the list of matching messages in the stream.
+
+
+        """
+        params = {
+            'sid': stream_id,
+            'since': since,
+            'session_token': self._auth_session.session_token,
+            'key_manager_token': self._auth_session.key_manager_token,
+            'skip': skip,
+            'limit': limit
+        }
+        message_list = await self._messages_api.v4_stream_sid_message_get(**params)
```

Comment:
What about `self._messages_api.v4_stream_sid_message_get(sid=stream_id, since=since, session_token=self._auth_session.session_token, ...)`
IMHO having a param dict which is then unpacked makes things more complicated.

Reply:
I would leave it as a param dict because for instance, `messages_api.v4_stream_sid_message_get` is defined as an instance of `Endpoint` callable class accepting the **kwargs parameters [Endpoint](https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/blob/2.0/symphony/bdk/gen/api_client.py#L751-L762)
And IMHO, it doesn't impact much to the complexity of the code

Reply:
I'd like to have someone else's opinion on that one, like @symphony-youri

Reply:
no strong opinion on that because it is internal and not exposed to users (and I have tried to quickly search for guidelines on how to be idiomatic on this and did not find anything to help me decide)

We have other places where add parameters conditionally (for attachments) so maybe we still need a map?
I would perhaps have a small method to reuse to avoid repeating the session_token part everywhere?